client/my-sites/pages/main.jsx

```diff
@@ -103,14 +114,26 @@ module.exports = React.createClass( {
 				</NavItem>
 			);
 		} );
-	},
+	}
 
 	_setWarning( selectedSite ) {
+		const { translate } = this.props;
 		if ( selectedSite && selectedSite.jetpack && ! selectedSite.hasMinimumJetpackVersion ) {
 			notices.warning(
-				this.translate( 'Jetpack %(version)s is required to take full advantage of all page editing features.', { args: { version: config( 'jetpack_min_version' ) } } ),
-				{ button: this.translate( 'Update now' ), href: selectedSite.options.admin_url + 'plugins.php?plugin_status=upgrade' }
+				translate( 'Jetpack %(version)s is required to take full advantage of all page editing features.', {
+					args: { version: config( 'jetpack_min_version' ) }
+				} ),
+				{ button: translate( 'Update now' ), href: selectedSite.options.admin_url + 'plugins.php?plugin_status=upgrade' }
 			);
 		}
 	}
-} );
+}
+
+const mapState = state => {
```

Comment:
As much as I personally prefer dropping the brackets, we're aiming for consistency, so args should be wrapped: `= ( state ) =>`.[^1] Also, might I suggest (my own personal preference) the shorthand version?

```
( state ) => ( {
  site: …,
  siteId: …,
} )
```

:)

[^1]: I think some exceptions are accepted, e.g. for curried functions (think Redux middleware).